Lib/tarfile.py

```diff
@@ -2567,7 +2568,26 @@ def chmod(self, tarinfo, targetpath):
         if tarinfo.mode is None:
             return
         try:
-            os.chmod(targetpath, tarinfo.mode)
+            try:
+                os.chmod(targetpath, tarinfo.mode)
+            except OSError as exc1:
+                # gh-108948: On FreeBSD, chmod() fails when trying to set the
+                # sticky bit on a file as non-root. On other platforms, the bit
+                # is silently ignored if it cannot be set. We make FreeBSD
+                # behave like other platforms by catching the error and trying
+                # again without the sticky bit.
+                if (hasattr(errno, "EFTYPE") and exc1.errno == errno.EFTYPE
+                    and (tarinfo.mode & stat.S_ISVTX)):
+                    try:
+                        # Retry without the sticky bit
+                        os.chmod(targetpath, tarinfo.mode & ~stat.S_ISVTX)
+                    except OSError as exc2:
+                        # The error from the second attempt is the direct cause
+                        # of the ExtractError, but we keep the original error
+                        # around for good information.
+                        raise exc2 from exc1
```

Comment:
Nested try/except are complicated. I suggest adding this comment for readability.

```suggestion
                        raise exc2 from exc1

                    # the second chmod() call without sticky bit succeeded
```

Reply:
Done